enterprise/frontend/src/metabase-enterprise/remote_sync/components/GitSyncControls/GitSyncControls.tsx

```diff
@@ -0,0 +1,237 @@
+    return buildCollectionTree(synced);
+  }, [collections]);
+
+  const isSwitchingBranch = !!nextBranch;
+  const isDirty = !!(dirtyData?.dirty && dirtyData.dirty.length > 0);
+  const isLoading = isSyncTaskRunning || isSwitchingBranch || isImporting;
+
+  const changeBranch = useCallback(
+    async (branch: string | null, isNewBranch?: boolean) => {
+      if (branch == null) {
+        console.warn("Trying to switch to null branch");
+        return;
+      }
+
+      if (!isNewBranch) {
+        await importChanges({ branch });
+
+        trackBranchSwitched({
+          triggeredFrom: "app-bar",
+        });
+      }
+
+      setNextBranch(null);
+    },
+    [importChanges],
+  );
+
+  const handleBranchSelect = useCallback(
+    async (branch: string, isNewBranch?: boolean) => {
+      try {
+        if (branch === currentBranch) {
+          return;
+        }
+
+        setNextBranch(branch);
+
+        const freshDirtyData = await refetchDirty().unwrap();
+        const hasDirtyChanges = freshDirtyData.dirty.length > 0;
+
+        if (hasDirtyChanges && !isNewBranch) {
+          setSyncConflictVariant("switch-branch");
+        } else {
+          await changeBranch(branch, isNewBranch);
+          setNextBranch(null);
+        }
+      } catch {
+        sendToast({
+          icon: "warning",
+          toastColor: "error",
+          message: t`Sorry, we were unable to switch branches.`,
+        });
+        setNextBranch(null);
+      }
+    },
+    [currentBranch, changeBranch, refetchDirty, sendToast],
+  );
+
+  const handlePullClick = useCallback(async () => {
+    if (!currentBranch) {
+      return;
+    }
+
+    try {
+      await importChanges({ branch: currentBranch }).unwrap();
+
+      trackPullChanges({
+        triggeredFrom: "app-bar",
+        force: false,
+      });
+
+      sendToast({
+        message: t`Your branch is now up to date with remote`,
+        icon: "check",
+      });
+    } catch (error) {
+      const { hasConflict, errorMessage } = parseSyncError(error as SyncError);
+
+      if (hasConflict) {
+        setSyncConflictVariant("pull");
+        return;
+      }
+
+      sendToast({
+        message: errorMessage || t`Failed to pull from remote`,
+        icon: "warning",
+      });
+    }
+  }, [currentBranch, importChanges, sendToast]);
+
+  const handlePushClick = useCallback(() => {
+    setShowPushModal(true);
+  }, []);
+
+  const handleClosePushModal = useCallback(() => {
+    setShowPushModal(false);
+  }, []);
+
+  const handleCloseSyncConflictModal = useCallback(() => {
+    setSyncConflictVariant(undefined);
+    setNextBranch(null);
+  }, []);
+
+  // Don't render if remote sync is not enabled, user is not admin, or not in read-write mode
+  if (
+    !isRemoteSyncEnabled ||
+    !isAdmin ||
+    !currentBranch ||
+    syncType !== "read-write"
+  ) {
+    return null;
+  }
+
+  return (
+    <>
+      <div className={styles.pill} data-testid="git-sync-controls">
```

Comment:
this is pretty special, but maybe Mantine's `<Button.Group />` would work instead of something so bespoke?

https://mantine.dev/core/button/#buttongroup